lib/util/strmap.ts

```diff
@@ -0,0 +1,80 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+export = StrMap;
+
+class StrMap<T> {
+    private _size: number = 0;
+    private _map: {[key: string]: T} = {};
+
+    get size(): number {
+        return this._size;
+    }
+
+    clear(): void {
+        this._size = 0;
+        this._map = {};
+    }
+
+    set(key: string, val: T): void {
+        var mkey = mangle(key);
+        if (!(mkey in this._map)) {
+           ++this._size;
+        }
+        this._map[mkey] = val;
+    }
+
+    get(key: string): T {
+        return this._map[mangle(key)];
+    }
+
+    has(key: string): boolean {
+        return mangle(key) in this._map;
+    }
+
+    delete(key: string): void {
+        var mkey = mangle(key);
+        if (mkey in this._map) {
+            --this._size;
+        }
+        delete this._map[mkey];
+    }
+
+    keys(): string[] {
+        var res: string[] = [];
+        for (var t in this._map) {
+            res.push(demangle(t));
+        }
+        return res;
```

Comment:
I wouldn't recommend using `in` because of the potential of iterating items on the prototype of `this._map`.  I functional approach that I would recommend:

```TypeScript
return _.map(Object.keys(this._map), (k: string) => {
    return demangle(k);
});
```